ui/src/org/pentaho/di/ui/spoon/trans/TransGraph.java

```diff
@@ -4223,16 +4222,6 @@ public void openMapping( StepMeta stepMeta, int index ) {
         && ( stepMeta.getStepMetaInterface().getActiveReferencedObjectDescription() == null || index < 0 ) ) {
         TransMeta subTransMeta = subTrans.getTransMeta();
         referencedMeta = subTransMeta;
-        if ( stepMeta.getStepMetaInterface() instanceof MetaInjectMeta ) {
-          // Make sure we don't accidently overwrite this transformation so we'll remove the filename and objectId
-          // Modify the name so the users sees it's a result
-          subTransMeta.setFilename( null );
```

Comment:
It looks like this was added so that if someone "opens referenced file" while running a transformation in Spoon, it modifies the object shown in Spoon so they can't save and overwrite the original (executing) sub-transformation.

Is there a way that this can be kept, even with the step being moved to a plugin?